nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/DeleteMarkLogic.java

```diff
@@ -81,40 +75,33 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
      * @return
      */
     @Override
-    protected QueryBatchListener buildQueryBatchListener(final ProcessContext context, final ProcessSession session) {
-        return new NiFiDeleteListener(session).onFailure((batch, throwable) -> {
-            synchronized (session) {
-                getLogger().error("Error deleting batch", throwable);
-                for (String uri : batch.getItems()) {
-                    FlowFile flowFile = session.create();
-                    session.putAttribute(flowFile, CoreAttributes.FILENAME.key(), uri);
-                    session.transfer(flowFile, FAILURE);
+    protected QueryBatchListener buildQueryBatchListener(final ProcessContext context, final ProcessSession session,
+                                                         final FlowFile incomingFlowFile) {
+        return batch -> {
```

Comment:
This is now simpler - if the deletion succeeds, then send a new FlowFile for each URI to `SUCCESS`. Otherwise, send a new FlowFile for each URI to `FAILURE`.